src/Neo.Extensions/Collections/HashSetExtensions.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// HashSetExtensions.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using System.Collections.Generic;
+
+namespace Neo.Extensions
+{
+    public static class HashSetExtensions
+    {
+        public static void Remove<T>(this HashSet<T> set, ISet<T> other)
+        {
+            if (set.Count > other.Count)
+            {
+                set.ExceptWith(other);
+            }
+            else
+            {
+                set.RemoveWhere(u => other.Contains(u));
+            }
+        }
```

Comment:
( Does `set.Remove(other)` have any difference from `set.ExceptWith(other)`, even when `set.Count <= other.Count`? )

Reply:
No clue. I'm just moving the method to appropriate namespaces or libraries.